### ISSUE TITLE
Hard to switch tabs in Android tablet

### ISSUE THREAD

**Issue by richb-hanover** [scripting/Scripting-News#287]
I have been reading Scripting News for a couple years on my Amazon Fire HD 8 (10th generation) running Android version 9 using Brave browser (version 1.61.109) 

Recently, I find I have a hard time changing/clicking new tabs on the main page. I expect to be able to tap to move to/select the desired tab. What actually happens is difficult to describe but is one of these behaviors:

- I almost always have to tap a tab twice to switch
- Sometimes I have to tap many more times.
- Maybe it helps to "double-tap" where I tap, wait bit, then tap again (about the time between beats for Stayin' Alive by the Bee Gees)
- In all cases, I simply farble around, clicking as long as necessary to get to the new tab

I have not determined a pattern to what triggers any of these behaviors. I have also tried varying tap-lengths from brief to holding down for a second. 

To rule out hardware problems, I made two tests, both work as expected, and seem to rule out any hardware problem in my tablet.

1. Clicking any of the underlined links in the body of the page works as expected: A tap will go to the desired page; long-clicking opens the context menu that allows me to open in a new tab, etc. 

2. I used archive.org to look at Scripting News in the past. The tabs work exactly as expected. I chose December 20, 2022 at 05:31:49 (https://web.archive.org/web/20221221053149/http://scripting.com/) . 

I started noticing this around the time the newest Scripting News came out; I remember some mention of making all the tab/menu handling the same across all the programs and I wonder if this behavior is related.

What other debugging information could I provide? Many thanks.

**Comment by scripting** [comment]
@richb-hanover -- thanks for the very thorough report. 

It sounds like the problems were introduced when I did a rewrite of the tab code, and it's probably due to some code catching the click before it gets to the tab code. Once found, it's easy to fix.

A couple of things to try --

1. Open the JavaScript console and then click on tabs and see if any error messages show up.
2. Try it with another browser, or try it with Brave on another machine. That would help determine if the issue is with Android or with Brave. I don't have an Android table to test with and don't use Brave.

I'll post a link to this issue on Scripting News, to see if other people have input. 

And thanks again for the thorough report. :-)

**Comment by richb-hanover** [comment]
**Update:** 

* I see the same behavior (difficulty switching tabs) in Chrome 106.0.5249.126 on my ancient Asus Nexus 7 running Android 6.0.1.
* I see the same behavior in Chrome 120.0.6099.119 running on an iPad running iOS 15.5 (both recent Chrome and iOS) 
* I will need some time to figure out how to attach a debugger to my tablet(s) for debugging. (I have discovered decent documentation - I need to work this in with other projects.)
* Thanks for the mention on Scripting News!

**Comment by richb-hanover** [comment]
You're welcome. My philosophy on this kind of stuff is expressed here: https://randomneuronsfiring.com/wpntfiiwdkib/

**Comment by scripting** [comment]
That's a great philosophy. Along a similar line, I've long felt they should teach every college grad how to write a good bug report. 

http://scripting.com/stories/2010/06/30/writingGoodBugReports.html

Your bug reports are great. I have been occupied doing other things, but when I can swing around to this I'm sure we'll be able zero in on the problem right away.

**Comment by scotthansonde** [comment]
I don't have an Android tablet, but I do see similar behavior on on my iPad (iPadOS 17.1.2) in both Chrome 120.0.6099.119 and Safari. Safari seems to catch the taps more often than Chrome, but it does miss one sometimes.

When I have a spare moment I'll to get a debugger going to see if there are any messages in the console.

**Comment by scripting** [comment]
I just tried it on my iPad in both Chrome and Safari, the performance was as it is on my desktop system, it works as it's supposed to.

**Comment by scripting** [comment]
Also please try it with this, a new version of the software, same content.

https://news.lucky.wtf/

**Comment by richb-hanover** [comment]
This version is better.

* Every tap switches to the new tab (this is progress!)
* But... There's a half-second to two-second delay between the tap and any visual change that confirms the tap. This is the sequence that I notice:
   - Tap 
   - Pause 0.5 to two seconds with no visual change. Then in rapid succession, I see:
      - Outline of new tab surrounding the proper title
      - Brief flash of blue to highlight the tab
      - Content appears below

Thanks!

**Comment by richb-hanover** [comment]
**Update:** Today, I went to news.lucky.wtf on my Fire tablet with Brave browser and it appears the tab behavior has changed, mostly for the better.

1. Tapping a tab _always_ switches to the desired tab. (This is a definite improvement.) It may take a bit of time to occur (see item 3 below) 
2. _Sometimes_ tapping a tab will immediately give a light blue background behind the text of the tab. That remains until the content gets loaded. (This is a great behavior. It gives immediate feedback that the tap was recognized and it's reasonable / understandable that the content might take some time to arrive.)
3. _Other times_, a tap gives no feedback for "a beat" (about a second?), then a brief flash of blue background behind the tab which then disappears, then a wait while the content downloads. 
4. _Nearly always,_ the link in the URL bar will change instantly to show the recently-tapped tab

Using my old (slower) Asus Nexus 7 tablet with Chrome:

* It works substantially as described above. 
* Most taps give item number 3. Item number 2 (immediate blue background feedback) is rare. 
* There's a much longer pause between the tap and the flash of blue background (2-4 seconds)
* _Most of the time_ the URL (item 4) changes immediately, but sometimes it takes a longer time, and occasionally it happens after the flash of the tab's background...

Using my iPhone 14 with either Chrome or Firefox:

* Tabs always switch as expected (Item number 2 above)
* I was surprised that tabs have a grey background, not a light blue highlight when switching.

I have not had time to figure out invoking the debugger to look for error messages in the browser console. Thanks again.

**Comment by scripting** [comment]
First @richb-hanover -- thanks for sticking with this. 

As I read your outline I was thinking, I wonder what "slow" looks like on his system.

So I thought I'd start by doing a demo of how news.lucky.wtf performs on my system.

https://github.com/scripting/Scripting-News/assets/1686843/d225d932-7f67-441a-a0d0-5b3710001541

**Comment by richb-hanover** [comment]
The attached video is exactly what it looks like on my desktop computer. I judge its speed to be completely acceptable. 

I need to figure out how to make a screen recording of an Android tablet where taps are "evident" to show what I'm seeing. I need to work on other projects today, but will look for time for a screen recording soon. Thanks again.

**Comment by scripting** [comment]
thanks, no rush. i've asked some other people to come take a look at this too.

**Comment by scripting** [comment]
@fmfernandes -- i haven't seen the brief blue flash. could you do a little video to demo?

**Comment by scripting** [comment]
@fmfernandes -- thanks -- i may want to get rid of the focus-setting. now i know what to look for. ;-)

**Comment by richb-hanover** [comment]
I figured out how to do screen recording. The attached video shows `news.lucky.wtf` as of 8:50pm on 15 Jan 2024 running on an Android Fire tablet in the Brave browser. Note that:

- a tap on some tabs instantly shows a blue background. This is great. It gives feedback that the tap was recognized. (See the "All" and "Politics" tabs, part way through)
- a tap on other tabs may take a half second to a couple seconds to be recognized in the GUI by drawing the tab border / displaying a blue background. Although 
- This is better than the production site (see below) because a tap always switches tabs

https://github.com/scripting/Scripting-News/assets/1094930/dd8dc7b0-69ec-4a39-8e0e-fc262ccf7f0d

For comparison, here's the production `scripting.com` site. Note the difficulty switching tabs - frequently requiring a double-tap to switch tabs:

https://github.com/scripting/Scripting-News/assets/1094930/56c7957f-7b97-4bb9-96a8-00834b442612

**Comment by richb-hanover** [comment]
@scripting I wonder if there's an update. (I updated the video links so they may appear now.) 

I continue to use news.lucky.wtf, where a single tap on a tab always switches to the tab. This remains different from the production scripting.com site, where I have to "tease" my Android tablet to switch.

I realize that it's not a vote, but I would be happy to see the production site switch to the code that runs news.lucky.wtf. Thanks as always.

**Comment by scripting** [comment]
I was working on something completely different today and saw the problem with tab-clicking. 

It happened in Chrome on my Mac in a mode where it emulates an iPad Mini. 

My first theory was there was another click handler somewhere that was catching the click and not propagating it. 

There aren't good ways to trap this in the debugger, but my limited ability to check this out via searching in source code said this wasn't it.

I looked for things I could turn off to see if the problem went away, and in that I seem to have found a conflict. 

If I commented out the code that adds a tooltip to the tab object then I no longer had any problems with clicking. 

Then I tried something less drastic. Instead of putting the tooltip on the whole tab, I just put it on the icon at the beginning of the tab. Again no problem. 

So here's how you can help.

1. I fixed the problem on the [scripting.com](http://scripting.com) home page.
2. So go there, on a tablet, or an emulated one, preferably something where it failed before, and try it now.

Report results here:

a) fixed
b) might be fixed
c) not fixed

**Comment by scripting** [comment]
I decided to make a stronger test, I commented out the code that adds the tooltip. That should make this a stronger test.

**Comment by richb-hanover** [comment]
I am sad to say that it is "c) not fixed". It does not appear to be materially different from my original report.

Here's how I tested at 2:40pm ET on 7 Mar 2024: 

I rebooted the same Fire 8 tablet as in the original report. I turned on Settings -> Device -> Developer -> Show taps to confirm that the OS registered my taps.

I then open `scripting.com` in the Brave browser.  I reloaded the page for good measure. Less than half the time taps are recognized. Frequently, I can tap, double-tap, tap-wait-a-beat-tap with no effect. I can usually get to a new tab by repeated fussing, but as I noted above, it is not materially different from the original report.

HOWEVER... `news.lucky.wtf` continues to work well. _Every_ tap on a tab is recognized. (It does take a while to retrieve the contents of the page, but the tab gives good feedback that the switch is happening.)

What other info could I provide? Thanks as always

**Comment by scripting** [comment]
@richb-hanover -- thanks for the report.

the good news is that i saw the problem here today. there's nothing i can do to fix the problem until i can reproduce it.

thanks for sticking with it.

**Comment by richb-hanover** [comment]
I don't mean to perseverate, but I want to be clear that the `news.lucky.wtf` tabs work perfectly in all my test cases (Android tablet, iPhone, desktop browser). Thanks.

**Comment by scripting** [comment]
I did a bit more research in trying to answer this question -- why would this happen on a mobile device and not on a website accessed from a desktop computer (assuming that's true for other people, it is true for me).

it turns out that the browser has special events for mobile devices because clicking is different there. they are called touchstart, touchend, touchmove. and they have another set of events that capture both classes, pointerdown, pointerup, pointermove. chatgpt recommends trying using "pointerdown" in place of "click" and seeing if that helps. as far as I can tell, it doesn't make a difference. the tabs still are only sometimes responsive.

**Comment by scripting** [comment]
okay so the next thing i want to look at is to see what's the difference between news.scripting.com and news.lucky.wtf.

there is a difference. both use the filter.js feature in pagepark, and a new feature in pagepark that hooks directly into FeedLand news products. so this is what's in filter.js on news.scripting.com

const newsProductOptions = {
	urlServer: "https://feedland.social/newsproduct",
	urlApp: "http://scripting.com/code/newsproducthome/index.html",
	urlNewsProductSpec: "http://scripting.com/code/newsproducthome/newsscriptingcom.json"
	};
options.runNewsProduct (newsProductOptions);

and this is what's in filter.js on news.lucky.wtf.

const newsProductOptions = {
	urlServer: "https://feedland.com/newsproduct",
	urlApp: "http://scripting.com/code/newsproducthome2/index.html",
	urlNewsProductSpec: "http://scripting.com/code/newsproducthome/newsscriptingcom2.json"
	};
options.runNewsProduct (newsProductOptions);

**Comment by scripting** [comment]
Reviewing the differences, 

1. urlServer -- one is being rendered by feedland.social (a new server) and the other is rendered by feedland.com. these are running the same software (feedland.com is one version behind the feedland.social, we should fix that). 
2. urlApp -- they're using different versions of the newsProductHome software.
3. urlNewsProductSpec -- they are defined by different specs. 

Now it's really hard for me to imagine that there are any real differences between the different versions and software, but there does seem to be a behavior difference. So I'm going to dive into that little cave and see if anything interesting shows up.

**Comment by scripting** [comment]
Before I dive too deep into this -- anyone who's tuned in -- can you try clicking on the tabs here:

https://news.scripting.com/

And say whether clicking works every time, or if you might have to click several times to get a tab to change.

Thanks!

**Comment by scripting** [comment]
I just did a text comparison between the two different versions, and the only difference I can see is that they're using different FeedLand instances to do the rendering of the news product. 

In the specs there are text and styling differences, and the order of the tabs changed, but it's really hard to imagine those having any impact on tab-clicking.

**Comment by scripting** [comment]
Okay here's another test.

http://scripting.com/code/tabsmanager2/testing/

Please click on that link, on a tablet system, and click on the tabs.

Same as before, does it work, or not.

There is no good or bad answer. It's just a test. 

Thanks! :-)

**Comment by grantdunham** [comment]
Using a Pixel 6 phone - Android 14; Chrome browser.
The page opens and the tabs work on first click.
The page size does not fit a phone - it spills to the right.
The tabs along the top - just 4 of them. No Bloggers or Dave tabs.

**Comment by jon-ault** [comment]
Works OK for me. Tabs took 1-2 seconds to populate but only had to click each one once. Using Safari on a  iPhone 15 Pro Max running iOS 17.3.1.

**Comment by richb-hanover** [comment]
Using Amazon Fire 8 tablet (same as original post) and Brave browser. This page looks good:

- Tap is always recognized, and always switches to desired content (after longish load time)
- Sometimes there's an immediate blue background, sometimes only an outline
- Here's a video of how it worked:
 

https://github.com/scripting/Scripting-News/assets/1094930/c9233442-7c9c-4ae3-96c7-e81fe777e5fc

(To help me remember next time, here's how I capture the video: 1) Turn on Settings -> Device -> Developer -> Show taps; 2) I used Screen Recorder Professional 1.0.0.0; 3) Open the app, this puts a little orange gizmo on the right edge; 4) tap the red dot to begin recording; 5) Make sure the screen orientation is as desired - recording retains original orientation; 6) when complete, click the orange gizmo, then hit two-parallel-bars; 7) go to the Screen Recorder Professional app; 8) touch the hamburger icon for the desired video on the right and choose Share; 9) use mail to send it to myself

**Comment by gwthompson** [comment]
@scripting -- Using an iPad Pro 12.9 the page opens and the tabs all work on first click and continue to work as I refresh the page and switch tabs.

Using a Lenovo Tab M10 tablet running Android 10 the page opens and the tabs all work on first click.

Using an iPhone 13 mini the page opens and the first 4 tabs are visible and all work on first click.

**Comment by MitchWagnerSD** [comment]
"All," "Politics," and "Tech" worked with one tap each. Had to tap "Podcasts" twice the first time I tried it, but after that performance was fine.

**Comment by scripting** [comment]
Please make it clear up front if it did or didn't work. The rest of it may be helpful insofar as it says what kind of tablet you're using and what browser. The rest of it is not helpful.

**Comment by MitchWagnerSD** [comment]
I did about 10 taps and it worked 9 times out of 10.

**Comment by moosebegab** [comment]
Works on ipad mini 6 / os 17.0.3 / chrome browser
Works on Motorola One 5G Ace / android 11 / chrome browser

> Okay here's another test.
> 
> http://scripting.com/code/tabsmanager2/testing/
> 
> Please click on that link, on a tablet system, and click on the tabs.
> 
> Same as before, does it work, or not.
> 
> Say what kind of device you were using.
> 
> There is no good or bad answer. It's just a test.
> 
> Thanks! :-)

**Comment by troutcolor** [comment]
All the tabs work fine here after many tries. 
iPadOS 17.3.1  iPad (8th generation)

cheers

john

**Comment by richb-hanover** [comment]
Additional Info: Going to http://scripting.com/code/tabsmanager2/testing/ with these devices worked as expected - no extra taps needed:

- Fire 7 tablet (older generation) - using native Silk browser
- Asus Nexus 7 tablet (quite old - 2013?) - using Brave browser

**Comment by danbarnes** [comment]
iPhone 14 Plus, up to date.

tapped on tabs,  right to left.

tapped tab > resulting tab:

podcasts > podcasts. ✔️
tech > tech  ✔️
politics > politics ✔️
but…
all > politics ❌

**Comment by RolHje** [comment]
Mac mini M1
Tabs work as expected, slight delay

**Comment by jdmarch** [comment]
Pixel 7, all good

**Comment by WSeidlich** [comment]
Moto G5 2023 tabs worked fine several times no problems. Thanks.

**Comment by scripting** [comment]
Okay it sounds like this version of the tabs works.

**Comment by LilacHammer6** [comment]
HP Envy Everything works. Clippers lost 67 - 102

**Comment by FLCoop** [comment]
Works fine on my iPad pro

**Comment by scripting** [comment]
### Next test

http://scripting.com/code/testing/tabstest/

This test is simpler, it just displays a picture in each tab.   

There is no good or bad result. If it doesn't work, we've gotten closer to finding the problem. If it does, then we know to look elsewhere. 

### How to

Click on each tab in turn, observe if a new picture has shown up below, and then click the next tab.

And then come back here and say if it worked or it didn't. 

Mention the device and browser you're using for the text.

### Thanks

And thanks for your help! :smile:

**Comment by moosebegab** [comment]
Tabs with pictures
Works on ipad mini 6 / os 17.0.3 / chrome browser / safari browser
Works on Motorola One 5G Ace / android 11 / chrome browser

**Comment by danbarnes** [comment]
iphone 14 plus, vivaldi (chromium) browser

TL;DR no problems

i first tapped tabs in order right to left, then tapped tabs at random.

each tab always displayed a picture. any given tab always displayed the
same picture.

**Comment by RolHje** [comment]
Mac mini M1, Sonoma 14.4, Safari
Displays different picture for every tab

**Comment by jon-ault** [comment]
Worked on iPhone 15 Pro Max, Safari, iOS 17.3.1.

**Comment by grantdunham** [comment]
Everything works on my Pixel in Chrome.

Grant

33 Hart Cres.
Whitehorse, YT
Y1A 4R2
(867) 633-3243 (home)
(867) 333-3243 (cell)

**Comment by jon-ault** [comment]
> Worked on iPhone 15 Pro Max, Safari, iOS 17.3.1.

Whoops, I was wrong. I didn’t realize it until I looked at it on my Mac, but on the iPhone the last two tabs are missing (Mets and Burns); even rotated to landscape they don’t appear.

**Comment by danbarnes** [comment]
I saw that someone reported missing tabs on their iPhone, so I went back
and made this screenshot on mine:

**Comment by richb-hanover** [comment]
Updated test results using http://scripting.com/code/testing/tabstest/ **TL;DR: works exactly right for all tests.**

Details:
- Works on all tested devices - Fire 8 (10th generation, Brave browser); Asus Nexus 7 (Android 6.0.1, Brave browser); Fire 7 (7th generation; Silk browser); ZTE Bravo 2 phone, Brave browser

- Works on all tested browsers on macOS (Firefox, Chrome, Safari, Brave)

- Tiny difference between tablets and desktop browsers:
   - On tablets, all the taps on tabs produced this visual sequence: immediate blue background behind tapped tab, near instantaneous outline of the tab itself; content (the picture) appeared nearly instantaneously as well. Working **perfectly**
   - On desktop browsers, it is identical except that I _never_ see the blue highlight behind the tab. This doesn't matter to the user experience. I am only reporting that the behavior is slightly different between these two cases.
   
THANK YOU FOR TRACKING THIS DOWN!

**Comment by mvopat** [comment]
It works for me on the iPhone 13 (with the latest operating system iOS 17.3.1)

**Comment by WSeidlich** [comment]
Tabs with pictures works w/ moto G5 2023 with duck duck go.

**Comment by scripting** [comment]
It seems like the second test is perfect. Everyone so far is reporting perfection.

If this keeps up there will be another test tomorrow that replaces one of the pictures with a timeline from FeedLand.

That will help determine if there's interference between the two. If the click meant for the tab is getting captured somehow by the FL bits.

**Comment by ttepasse** [comment]
I’m sorry to say I’ve got your next problems to tackle, reporting them here, because I don’t know where else: **Keyboard Controls** and **Accessibility**.

# Keyboard Controls

Ever since the UIs became stable in 90s users expect a common interaction model for tabbed views:

- You can tab from the previous focusable control to the tabbed view. The focus lands on the selected tab.
- You can navigate between tabs with the arrow keys.
- For an *manual tab view* you’d navigate the focus and then activate the tab and its tab panel with Space or Enter.
- For an *automatic tab view* the act of navigating automatically activates the tab.
- Tabbing out of the list of tabs brings you to the next focusable element either in the currently active tab panel when there is one or to the next focusable element on the page.

Your current example does nothing of that. It already fails when trying to tab to the tab list.

A view into the HTML/DOM tells us why: your list of tabs doesn’t use focusable elements like buttons or activatable links, not do you use the [`tabindex` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex?retiredLocale=de) to make plain elements focusable. What you can’t focus, you can’t activate via keyboard.

Keyboard controls are not just for lazy mousers like me, but rather important for people who lack the dexterity in their hand to control the mouse or the vision to see their mouse pointer. Which brings me to my second point:

# Accessibility 

Your HTML doesn’t tell the browser that it takes the role of a tabbed view. Hence assistive technology like screenreader can’t tell the user that it is a tab view, which thing is a tab, which tab is selected, etc.

WAI-ARIA has a set of attributes which annotate HTML elements which UI role they inhabit:

- A list of tabs should be annotated with `role="tablist"`.
- The individual tabs should be annotated with `role="tab"` and each tab element should use the attribute `aria-controls="id"` to identify the corresponding tab panel.
- The currently selected tab should use `aria-selected="true"` and the others `false`.
- Tab panels are annotated with `role="tabpanel` and should possibly indicate with `aria-labelledby` which tab acts as their label/heading.

If you're testing a web page (on a Mac) with the VoiceOver screenreader or the [Accessibility Inspector](https://developer.apple.com/documentation/accessibility/inspecting-the-accessibility-of-screens) those annotations make a difference, telling the Voice-only user what is what.

# Inspiration & Running Code

The WAI has guidelines and examples exactly for tabbed view with running code:

- [Tabs Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/)
- [Example of Tabs with Manual Activation](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-manual/)
- [Example of Tabs with Automatic Activation](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/)

Heydon Pickering in his book and blog *Inclusive Components* walks the reader through the process of building an tabbed interface, again with working code:

- [Tabbed Interfaces - Inclusive Components](https://inclusive-components.design/tabbed-interfaces/)

Hope that helps.

**Comment by gillesenvrac** [comment]
If I open the link on my iPhone 13, when in landscape, I see the 5 tabs. But when I use it in portrait mode I miss the last two tabs.

**Comment by danbarnes** [comment]
Similarly on my iPhone 14 Plus:

1) Open test page in portrait, get four tabs.
2) Reorient to landscape and then refresh the page,  get six tabs.
3) Reorient to portrait and then refresh the page, get four tabs.

**Comment by scripting** [comment]
Here's the next test. It's the same as the previous test, except one of the tabs is a FeedLand timeline. 

### How to

1. Go to this page.
2. Click on each tab in turn from left to right.
3. Then starting at the last tab go from right to left. 
4. Do the back and forth a few times.
5. Try clicking on random tabs.

### Question

Did you have to click on any tab more than once to get it to change?

Also include info about browser and device.

**Comment by scripting** [comment]
@danbarnes -- this isn't a test of whether or not the tabs are visible on a screen, I've tried to explain carefully what we're testing for. So I'm deleting your second comment as I deleted your first.

**Comment by mokuyo** [comment]
iPhone 15 Pro, iOS 17.4
Displayed tabs are responsive through repeated and random clicks.

**Comment by scripting** [comment]
I started a new thread for the fourth test and beyond. 

https://github.com/scripting/Scripting-News/issues/295#issuecomment-1987264407

Please stay with it. I can move pretty fast if there are enough people trying to make it fail.

**Comment by richb-hanover** [comment]
Using http://scripting.com/code/testing/tabstest3/index.html, I saw interesting behavior for the River tab:

- For all tablet devices tabs, tabs always switched after a single tap (see https://github.com/scripting/Scripting-News/issues/287#issuecomment-1986915087 for the four devices I tested) 

- For some devices, the River tab was highlighted immediately after the tap. It took a while for the content to show up (as always)

- For the other devices (two of the four), the River tab _did not highlight_ in any way until the content showed up, at which point the blue background flashed, the outline for the tab appeared, and the content for the pane appeared.

- As instructed, I clicked left-to-right, then back right-to-left. I think I noticed slightly different behavior for the River tab on the right-to-left tab. Let me know if it would be helpful to have more info

**Comment by moosebegab** [comment]
Tabs with pictures and Feedland timeline
all worked on first click.
Works on ipad mini 6 / os 17.0.3 / chrome browser / safari browser
Works on Motorola One 5G Ace / android 11 / chrome browser

**Comment by scripting** [comment]
the only question that matters is whether it ignores your click or responds to it.

it doesn't matter _how_ it responds to it. 

thanks for keeping the responses simple.

**Comment by scripting** [comment]
Your help is much appreciated! :smile:

Let's try this as a tab test.

https://news.scripting.com/

As with every other test...

1. Click a tab.
3. Ignore how long it takes to display the content.
4. Did it ignore your click? 
5. Click another tab.

Do this a bunch of times. 

Did it register your clicks or not?

Remember, forget everything else. All that matters is the tab clicking being heard by the software.

**Comment by grantdunham** [comment]
Pixel Phone - Android and Chrome. Worked.

**Comment by mokuyo** [comment]
iPhone 15 Pro, iOS 17.4
Every.click.works.

**Comment by danbarnes** [comment]
Worked

iPhone 14 Plus, Vivaldi (Chromium) browser

**Comment by Solo-500** [comment]
tabs work, paragraphs in News run too wide. iphone mini 13.
